sensor/common/deploymentenhancer/component.go

```diff
@@ -0,0 +1,144 @@
+package deploymentenhancer
+
+import (
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/set"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+	"github.com/stackrox/rox/sensor/common/store"
+)
+
+var (
+	log                 = logging.LoggerForModule()
+	deploymentQueueSize = 50 // TODO(ROX-21291): Configurable via env var
+)
+
+// The DeploymentEnhancer takes a list of Deployments and enhances them with all available information
+type DeploymentEnhancer struct {
+	responsesC       chan *message.ExpiringMessage
+	deploymentsQueue chan *central.DeploymentEnhancementRequest
+	storeProvider    store.Provider
+}
+
+// CreateEnhancer creates a new Enhancer
+func CreateEnhancer(provider store.Provider) common.SensorComponent {
+	return &DeploymentEnhancer{
+		responsesC:       make(chan *message.ExpiringMessage),
+		deploymentsQueue: make(chan *central.DeploymentEnhancementRequest, deploymentQueueSize),
+		storeProvider:    provider,
+	}
+}
+
+// ProcessMessage takes an incoming message and queues it for enhancement
+func (d *DeploymentEnhancer) ProcessMessage(msg *central.MsgToSensor) error {
+	toEnhance := msg.GetDeploymentEnhancementRequest()
+	if toEnhance == nil {
+		return nil
+	}
+	log.Debugf("Received message to process in DeploymentEnhancer: %v++", msg)
```

Comment:
++? I think you mean
```suggestion
	log.Debugf("Received message to process in DeploymentEnhancer: %+v", msg)
```

Reply:
Furthermore: 
```suggestion
	log.Debugf("Received message to process in DeploymentEnhancer: %v++", toEnhance)
```